doc/modules/grid_search.rst

```diff
@@ -150,6 +154,324 @@ increasing ``n_iter`` will always lead to a finer search.
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching optimal parameters with successive halving
+====================================================
+
+Scikit-learn also provides the :class:`HalvingGridSearchCV` and
+:class:`HalvingRandomSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving is an iterative selection process where all candidates (the
+parameter combinations) are evaluated with a small amount of resources at
+the first iteration. Only some of these candidates are selected for the next
+iteration, which will be allocated more resources. What defines a resource
+is typically the number of samples to train on, or the number of iterations
+in iterative algorithms like gradient boosting.
```

Comment:
I think we need a full-fledged example with `HistGradientBoostingClassifier` with `warn_start=True` and `HalvingRandomSearchCV` with `resource="max_iter"` and a fixed `n_candidates=30` that would compare the results with `RandomizedSearchCV` with `n_iter=30` to see the kind of speed up we could get.

Reply:
Alright. I did some tests and even without `warm_start` support it seems possible to benefit from max_iter so it would be good to have an example that compare the use of `resource='n_samples'` vs `resource='max_iter'` and discus the pros and cons of each approach (and compare to a `RandomizedSearchCV` baseline.